Source/WebInspectorUI/UserInterface/Views/FolderTreeElement.js

```diff
@@ -25,11 +25,36 @@
 
 WI.FolderTreeElement = class FolderTreeElement extends WI.GeneralTreeElement
 {
-    constructor(title, representedObject)
+    constructor(title, representedObject, id = null)
     {
         const classNames = [WI.FolderTreeElement.FolderIconStyleClassName];
         const subtitle = null;
         super(classNames, title, subtitle, representedObject, {hasChildren: true});
+
+        if (id) {
+            let settingName = "folder-tree-element-" + id;
+            const defaultValue = false;
+            this._expandedSetting = new WI.Setting(settingName, defaultValue);
+
+            if (this._expandedSetting.value)
+                this.expand();
+        }
+    }
+
+    collapse()
```

Comment:
Style: there should be a `// Public` comment here to help identify these as public member functions
```js
}

// Public

collapse()
{
```